### PR TITLE
feat(manage-live-stream): page manage live stream is ready for test

### PR DESCRIPTION
**Pull request by DiLiUA**
- refactored show service;
  - created and development functionality ManageLiveStreamComponent;
  - reworked WowzaCloudService;
  - reworked WowzaPlayerComponent;
  - added EventEmitter for cancell button;
  - removed myEventComponent;
  - removed WowzaCloudComponent;
  - removed WowzaCloudModule;

### CODE REVIEW

File: src/manage-live-stream/manage-live-stream.component.html [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<div class="custom-container">
+  <div class="top-part" *ngIf="currentShow">
+    <div class="show-title">{{currentShow.name}}</div>
+    <div class="date-performance">{{currentShow.timePerformance.start}}</div>
+  </div>
+
+  <div class="stream-container" *ngIf="currentShow && currentShow.wowza">
+    <div class="btns-container">
+      <button type="button" class="btn-with-border btn-orange small" [disabled]="isStartedStream" (click)="startLiveStream()">Start the live stream</button>
+      <button type="button" class="btn-with-border btn-orange small" [disabled]="!isStartedStream" (click)="stopLiveStream()">Stop the lives tream</button>
```
Comment (Ugmaxie) [inline src/manage-live-stream/manage-live-stream.component.html]: please correct a mistake in words

File: src/manage-live-stream/manage-live-stream.component.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+@Component({
+  selector: 'app-manage-live-stream',
+  templateUrl: './manage-live-stream.component.html',
+  styleUrls: ['./manage-live-stream.component.scss']
+})
+export class ManageLiveStreamComponent implements OnInit, OnDestroy {
+  subscriptionManager: Subscription = new Subscription();
+  currentShow: Show;
+  isStartedStream = false;
+  liveStreamId: string = null;
```
Comment (Ugmaxie) [inline src/manage-live-stream/manage-live-stream.component.ts]: type string, but null?

File: src/manage-live-stream/manage-live-stream.component.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+      .subscribe(stream => {
+        this.currentShow.wowza = stream;
+      }, err => {
+        console.error('something went wrong: ', err);
+      });
+
+    const getLiveStreamStateSubscription = this.wowzaCloudService.getLiveStreamState(this.liveStreamId)
+      .subscribe(res => {
+        this.isStartedStream = res.state === 'started' || res.state === 'starting';
+        console.log(this.currentShow);
```
Comment (Ugmaxie) [inline src/manage-live-stream/manage-live-stream.component.ts]: no console.log

File: src/manage-live-stream/manage-live-stream.component.ts [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+  }
+
+  startLiveStream(): void | undefined {
+    if (!this.currentShow || !this.liveStreamId || this.isStartedStream) {
+      return undefined;
+    }
+
+    const startLiveStreamSubscription = this.wowzaCloudService.startLiveStream(this.liveStreamId)
+      .subscribe(() => {
+        this.isStartedStream = true
```
Comment (Ugmaxie) [inline src/manage-live-stream/manage-live-stream.component.ts]: ;

File: src/shared/wowza-streaming-cloud/wowza-cloud.service.ts [hunk 5]
```diff
@@ -1,82 +1,131 @@
...
-    });
+  constructor(private http: HttpClient) {
   }
 
-  getStream(id: string): Observable<any> {
-    return this.http.get(`${this.url}live_streams/${id}`, this.setWowzaOptions())
-      .map(res => res.json().live_stream);
-  }
+  newLiveStream(params: NewStreamModel): Observable<WowzaResponse> {
+    // const rqstBody = {live_stream: {...newStreamObj.live_stream, ...params}};
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-cloud.service.ts]: no comments

File: src/shared/wowza-streaming-cloud/wowza-player/wowza-player.component.ts [hunk 6]
```diff
@@ -10,38 +12,46 @@ import { PlayerConfig } from './player-config.interface';
...
   }
 
   parsePlayerConfig(): PlayerConfig {
     return {
-      title: this._showParams.name,
-      description: this._showParams.description,
-      sourceURL: this._showParams.wowza.player_hls_playback_url,
+      title: this.showParams.name,
+      description: this.showParams.description,
+      // sourceURL: this.showParams.wowza.player_hls_playback_url,
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-player/wowza-player.component.ts]: no comments

File: src/shared/wowza-streaming-cloud/wowza-player/wowza-player.component.ts [hunk 7]
```diff
@@ -10,38 +12,46 @@ import { PlayerConfig } from './player-config.interface';
...
+      sourceURL: 'https://fe159f.entrypoint.cloud.wowza.com/app-77cf/ngrp:cfb45107_all/playlist.m3u8',
+      // should be deleted when we have Wowza licence
```
Comment (Ugmaxie) [inline src/shared/wowza-streaming-cloud/wowza-player/wowza-player.component.ts]: no comments

File: src/event-launch/event-launch.component.ts [hunk 8]
```diff
@@ -165,9 +161,9 @@ export class LaunchComponent implements OnInit, OnDestroy {
     this.launchEvent.timePerformance.start = moment(this.timePerformance.start).format('dddd, MMMM DD YYYY, h:mm:ss a');
     this.launchEvent.timePerformance.end = moment(this.timePerformance.end).format('dddd, MMMM DD YYYY, h:mm:ss a');
 
-    const newStreamSubscription = this.wowzaCloudService.newStream(this.wowzaObj)
+    const newStreamSubscription = this.wowzaCloudService.newLiveStream(this.wowzaObj)
       .subscribe(liveStream => {
-        this.launchEvent.wowza.id = liveStream.live_stream.id;
+        this.launchEvent.wowza.id = liveStream.id;
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.component.ts]: please avoid subscribe into another subscribe
something like that:

const newStreamSubscription = this.wowzaCloudService.newLiveStream(this.wowzaObj)
  .map(liveStream => liveStream.id)
  .mergeMap(liveStreamId => {
    this.launchEvent.wowza.id = liveStreamId;
    return this.eventService.saveNewEvent(this.launchEvent)
}).subscribe(res => {
    this.posterUploader.sendFiles(res.newId);
    this.router.navigate(['events']);
  }, err => {
    console.error('something went wrong: ', err);
  });

**Review by Ugmaxie — CHANGES_REQUESTED** [review]

File: src/shared/wowza-streaming-cloud/wowza-player/wowza-player.component.ts (hunk 6 above)
Comment (DiLiUA) [reply src/shared/wowza-streaming-cloud/wowza-player/wowza-player.component.ts]: this comment needs for future. When we will have a Wowza license we should use commented code.

**Review by Ugmaxie — APPROVED** [review]

**Pull request merged**